### PR TITLE
Add `<ReadMore>` component

### PR DESCRIPTION
**Pull request by delucis**
#### Description (required)

- Replaces our 📚 bullets with a `<ReadMore>` component that adds a stylized open book icon to text instead.

   Usage:

   ```diff
   - 📚 Learn more about [emoji](#emoji)
   + import ReadMore from '~/components/ReadMore.astro';
   +
   + <ReadMore>Learn more about [emoji](#emoji)</ReadMore>
   ```

- Updates our `<RecipeLinks>` component styles to match `<ReadMore>` more closely

- Updates a few instances of the 📚 pattern to use Starlight’s `<LinkCard>` component instead where that felt more appropriate

### CODE REVIEW

**Review by VoxelMC — APPROVED** [review]
Looks awesome to me, great work everyone!!

File: src/content/docs/en/guides/integrations-guide/vercel.mdx [hunk 1]
```diff
@@ -94,7 +95,7 @@ import vercel from '@astrojs/vercel/static';
 
 ## Usage
 
-📚 **[Read the full deployment guide here.](/en/guides/deploy/vercel/)**
+<ReadMore>**[Read the full deployment guide here.](/en/guides/deploy/vercel/)**</ReadMore>
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/vercel.mdx:98]:
```suggestion
<ReadMore>Find out more about [deploying your project to Vercel](/en/guides/deploy/vercel/).</ReadMore>
```

File: src/content/docs/en/guides/integrations-guide/markdoc.mdx [hunk 2]
```diff
@@ -290,7 +291,7 @@ export default defineMarkdocConfig({
 });
 ```
 
-📚 [Find all of Markdoc's built-in nodes and node attributes on their documentation.](https://markdoc.dev/docs/nodes#built-in-nodes)
+<ReadMore>[Find all of Markdoc's built-in nodes and node attributes on their documentation.](https://markdoc.dev/docs/nodes#built-in-nodes)</ReadMore>
```
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/markdoc.mdx:294]:
```suggestion
<ReadMore>Find all of the built-in nodes and node attributes on the [Markdoc nodes documentation](https://markdoc.dev/docs/nodes#built-in-nodes).</ReadMore>
```

File: src/content/docs/en/guides/typescript.mdx [hunk 3]
```diff
@@ -281,9 +281,11 @@ To see type errors in your editor, please make sure that you have the [Astro VS
 `astro check` checks all the files included in your TypeScript project. To check types within Svelte and Vue files, you can use the [`svelte-check`](https://www.npmjs.com/package/svelte-check) and the [`vue-tsc`](https://www.npmjs.com/package/vue-tsc) packages respectively.
 :::
 
-📚 Read more about [`.ts` file imports](/en/guides/imports/#typescript) in Astro.
+import ReadMore from '~/components/ReadMore.astro'
 
-📚 Read more about [TypeScript Configuration](https://www.typescriptlang.org/tsconfig/).
+<ReadMore>Read more about [`.ts` file imports](/en/guides/imports/#typescript) in Astro.</ReadMore>
+
+<ReadMore>Read more about [TypeScript Configuration](https://www.typescriptlang.org/tsconfig/).</ReadMore>
```
Comment (sarah11918) [inline src/content/docs/en/guides/typescript.mdx:288]:
```suggestion
<ReadMore>Read more about [`.ts` file imports](/en/guides/imports/#typescript) in Astro or [TypeScript Configuration](https://www.typescriptlang.org/tsconfig/).</ReadMore>
```

File: src/content/docs/en/install/auto.mdx [hunk 4]
```diff
@@ -133,8 +133,22 @@ Success! You are now ready to start building with Astro! 🥳
...
+    title="Deploy your site"
+    description="Learn how to build and deploy an Astro project to the web in our Deployment guide."
+    href="/en/guides/deploy/"
+  />
+  <LinkCard
+    title="Understand your codebase"
+    description="Learn more about Astro’s file structure in our Project Structure guide."
+    href="/en/core-concepts/project-structure/"
+  />
+</CardGrid>
```
Comment (sarah11918) [inline src/content/docs/en/install/auto.mdx:154]:
```suggestion
<CardGrid>
    <LinkCard
    title="Understand your codebase"
    description="Learn more about Astro’s file structure in our Project Structure guide."
    href="/en/core-concepts/project-structure/"
  />
  <LinkCard
    title="Add a framework"
    description="Extend Astro with support for React, Svelte, Tailwind and more in our Integrations guide."
    href="/en/guides/integrations-guide/"
  />
  <LinkCard
    title="Deploy your site"
    description="Learn how to build and deploy an Astro project to the web in our Deployment guide."
    href="/en/guides/deploy/"
  />
</CardGrid>
```

**Review by sarah11918 — APPROVED** [review]
Just left some nits! I think we can clean up a couple of those existing links to have better text describing the link, and I think I like switching up the order on the installation page!

File: src/content/docs/en/install/auto.mdx (hunk 4 above)
Comment (VoxelMC) [reply src/content/docs/en/install/auto.mdx:154]: I agree! Makes more sense in `my` head.

File: src/content/docs/en/guides/integrations-guide/markdoc.mdx (hunk 2 above)
Comment (VoxelMC) [reply src/content/docs/en/guides/integrations-guide/markdoc.mdx:294]: I was going to do this when I got to my laptop :P

Comment (delucis) [reply src/content/docs/en/guides/integrations-guide/markdoc.mdx:294]: Dont love having a “their” before “Markdoc” is mentioned. Also not sure if “their” is appropriate (who is the plural group? is “Markdoc” they?)

Going to go with this instead!

```suggestion
<ReadMore>See the [Markdoc nodes documentation](https://markdoc.dev/docs/nodes#built-in-nodes) to learn about all the built-in nodes and attributes.</ReadMore>
```

**Pull request merged**